streams/src/test/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImplTest.java

```diff
@@ -203,6 +204,78 @@ private void setupGetSensorTest(final Metrics metrics,
         replay(metrics);
     }
 
+    @Test
+    public void shouldGetSameThreadLevelSensor() {
+        final Metrics metrics = mock(Metrics.class);
+        final RecordingLevel recordingLevel = RecordingLevel.INFO;
+        setupGetSensorTest(metrics, THREAD_ID, recordingLevel);
+
+        final StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(metrics, CLIENT_ID, VERSION);
+
+        final Sensor preSensor = streamsMetrics.taskLevelSensor(THREAD_ID, TASK_ID, sensorName1, recordingLevel);
+        final Sensor afterSensor = streamsMetrics.taskLevelSensor(THREAD_ID, TASK_ID, sensorName1, recordingLevel);
+
+        final Sensor actualSensor = streamsMetrics.taskLevelSensor(THREAD_ID, TASK_ID, sensorName1, recordingLevel);
+
+        verify(metrics);
+        assertThat(actualSensor, is(equalToObject(sensor)));
+        assertThat(preSensor, is(equalToObject(afterSensor)));
+    }
```

Comment:
There several issues with this test:
- First of all the test fails.
- According to the name of the test you want to verify `threadLevelSensor()`, but you call `taskLevelSensor()`.
- Since the `Metrics` mock always returns the same sensor, it does not make sense to compare the sensors that are returned by the different calls to `threadLevelSensor()`. Such a verification will always be true. You should rather verify if method `sensor()` is not called on the `Metrics` mock. For example, the following two setups could replace `setupGetSensorTest()`:
```
    private void setupGetNewSensorTest(final Metrics metrics,
                                       final String level,
                                       final RecordingLevel recordingLevel) {
        final String fullSensorName = fullSensorName(level);
        expect(metrics.getSensor(fullSensorName)).andStubReturn(null);
        final Sensor[] parents = {};
        expect(metrics.sensor(fullSensorName, recordingLevel, parents)).andReturn(sensor);
        replay(metrics);
    }

    private void setupGetExistingSensorTest(final Metrics metrics,
                                            final String level,
                                            final RecordingLevel recordingLevel) {
        final String fullSensorName = fullSensorName(level);
        expect(metrics.getSensor(fullSensorName)).andStubReturn(sensor);
        replay(metrics);
    }
``` 

and the following two tests would replace `shouldGetTaskLevelSensor()`:

```
    @Test
    public void shouldGetNewThreadLevelSensor() {
        final Metrics metrics = mock(Metrics.class);
        final RecordingLevel recordingLevel = RecordingLevel.INFO;
        setupGetNewSensorTest(metrics, THREAD_ID, recordingLevel);
        final StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(metrics, CLIENT_ID, VERSION);

        final Sensor actualSensor = streamsMetrics.threadLevelSensor(THREAD_ID, sensorName1, recordingLevel);

        verify(metrics);
        assertThat(actualSensor, is(equalToObject(sensor)));
    }

    @Test
    public void shouldGetExistingThreadLevelSensor() {
        final Metrics metrics = mock(Metrics.class);
        final RecordingLevel recordingLevel = RecordingLevel.INFO;
        setupGetExistingSensorTest(metrics, THREAD_ID, recordingLevel);
        final StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(metrics, CLIENT_ID, VERSION);

        final Sensor actualSensor = streamsMetrics.threadLevelSensor(THREAD_ID, sensorName1, recordingLevel);

        verify(metrics);
        assertThat(actualSensor, is(equalToObject(sensor)));
    }
```

Similar is true for the other tests below.